app.js

```diff
@@ -1,18 +1,84 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const hbs = require("hbs");
 
 // require spotify-web-api-node package here:
 
+const SpotifyWebApi = require("spotify-web-api-node");
+
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
 
 // setting the spotify-api goes here:
 
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch((error) =>
+    console.log("Something went wrong when retrieving an access token", error)
+  );
+
 // Our routes go here:
 
-app.listen(3000, () => console.log('My Spotify project running on port 3000 🎧 🥁 🎸 🔊'));
+app.get("/", (request, response) => {
+  response.render("home");
+  // response.sendFile(__dirname + "/views/home.hbs");
+});
+
+app.get("/artist-search", (req, res) => {
+  spotifyApi
+    .searchArtists(req.query.artists)
+    .then((data) => {
+      console.log("The received data from the API: ", data.body.artists.items);
+      // ----> 'HERE'S WHAT WE WANT TO DO AFTER RECEIVING THE DATA FROM THE API'
+
+      let allArtist = data.body.artists.items;
+      res.render("artist-search-results", { allArtist });
+      //   res.send(data)
+    })
+    .catch((err) =>
+      console.log("The error while searching artists occurred: ", err)
+    );
+  //
+});
+
+app.get("/albums/:artistId", (req, res, next) => {
+  spotifyApi
+    .getArtistAlbums(req.params.artistId)
+    .then((data) => {
+      let allAlbums = data.body.items;
+      //   res.send(data)
+      res.render("albums", { allAlbums });
+    })
+    .catch((err) =>
+      console.log("The error while searching artist occurred", err)
+    );
+});
+
+app.get("/tracks/:id", (req, res) => {
+  spotifyApi
+    .getAlbumTracks(req.params.id)
+    .then((data) => {
+      console.log("The received data from the API: ", data.body.items[0].url);
+
+      //res.send(data)
+      res.render("tracks", { result: data.body });
```

Comment:
You could've passed `data.body.items` here so you wouldn't need to pass `result.items` on your view